src/matcher/plugin.py

```diff
@@ -336,4 +342,48 @@ def pytest_addoption(parser) -> None:
       , default=False
       )
 
+
+@pytest.hookimpl(trylast=True)
+def pytest_configure(config: pytest.Config):
+    """Register additional configuration."""
+    if config.getoption('--pm-reveal-unused-files'):
+        config.option.collectonly = True
+        reporter = UnusedFilesReporter()
+        config.pluginmanager.unregister(name='terminalreporter')
+        config.pluginmanager.register(reporter, 'terminalreporter')
+
+
+class UnusedFilesReporter:
+    """A reporter to reveal unused pattern files."""
+    def pytest_collection_finish(self, session: pytest.Session) -> None:
+        """Once test items have been collected, check for and show unused files."""
+        if not session.items:
+            return
+
+        for alg in [_try_cli_option, _try_ini_option]:
+            patterns_base_dir, _ = alg(session.items[0]._request)  # NOQA: SLF001
+            if patterns_base_dir:
+                break
+        else:
+            message = 'Failed to determine a patterns base directory'
+            raise pytest.UsageError(message)
+
+        known_extensions = '.out', '.err'
+
+        all_paths: list[pathlib.Path] = []
+        for p in patterns_base_dir.rglob('*'):
+            if p.is_file() and p.suffix in known_extensions:
+                all_paths.append(p.resolve())  # NOQA: PERF401
+
+        collected_paths: list[pathlib.Path] = []
+        for item in session.items:
+            for fixture, ext in zip((expected_out, expected_err), known_extensions):
+                if fixture.__name__ in item.fixturenames:
+                    filename = _make_expected_filename(item._request, ext)  # NOQA: SLF001
```

Comment:
Oh... accessing a private member is not really nice :() Would be great to find some other way if possible %()

Reply:
Yeah, unfortunately, I don't see any other way.  There's no such hook that takes an argument of `pytest.FixtureRequest` directly. I think they made the `_request` attribute private because it's not immediately available to all hooks.